pkg/apis/pipeline/v1beta1/pipeline_validation_test.go

```diff
@@ -2571,6 +2572,72 @@ func TestPipelineTasksExecutionStatus(t *testing.T) {
 	}
 }
 
+// TestIncompatibleAPIVersions exercises validation of fields that
+// require a specific feature gate version in order to work.
+func TestIncompatibleAPIVersions(t *testing.T) {
+	tests := []struct {
+		name            string
+		requiredVersion string
+		spec            PipelineSpec
+	}{{
+		name:            "matrix requires alpha - check tasks",
+		requiredVersion: "alpha",
+		spec: PipelineSpec{
+			Tasks: PipelineTaskList{{
+				Name:    "a-task",
+				TaskRef: &TaskRef{Name: "a-task"},
+				Matrix: []Param{{
+					Name: "a-param", Value: ArrayOrString{Type: ParamTypeArray, ArrayVal: []string{"foo", "bar"}},
+				}},
+			}},
+		},
+	}, {
+		name:            "matrix requires alpha - check finally tasks",
+		requiredVersion: "alpha",
+		spec: PipelineSpec{
+			Tasks: PipelineTaskList{{
+				Name:    "a-task",
+				TaskRef: &TaskRef{Name: "a-task"},
+			}},
+			Finally: PipelineTaskList{{
+				Name:    "b-task",
+				TaskRef: &TaskRef{Name: "b-task"},
+				Matrix: []Param{{
+					Name: "a-param", Value: ArrayOrString{Type: ParamTypeArray, ArrayVal: []string{"foo", "bar"}},
+				}},
+			}},
+		},
+	}}
+	versions := []string{"alpha", "stable"}
+	for _, tt := range tests {
+		for _, version := range versions {
+			testName := fmt.Sprintf("(using %s) %s", version, tt.name)
+			t.Run(testName, func(t *testing.T) {
+				ps := tt.spec
+				featureFlags, _ := config.NewFeatureFlagsFromMap(map[string]string{
+					"enable-api-fields": version,
+				})
+				cfg := &config.Config{
+					FeatureFlags: featureFlags,
+				}
+
+				ctx := config.ToContext(context.Background(), cfg)
+
+				ps.SetDefaults(ctx)
+				err := ps.Validate(ctx)
```

Comment:
NIT: suggest calling `validateMatrix` here instead of `validate` since the intention of this unit test is to validate `matrix` specific functionality.

Reply:
would prefer to have it in the pipelinespec validation, so i can check against both `tasks` and `finally`, and to confirm that the `validateMatrix` is used in the pipeline validation